src/backends/postgresql/session.cpp

```diff
@@ -29,6 +30,130 @@ void hard_exec(postgresql_session_backend & session_backend,
     postgresql_result(session_backend, PQexec(conn, query)).check_for_errors(errMsg);
 }
 
+// helper function to quote a string before sinding to PostgreSQL
+char * quote(PGconn * conn, const char *s, size_t len)
+{
+    int error_code;
+    char *retv = new char[2 * len + 3];
+    retv[0] = '\'';
+    size_t len_esc = PQescapeStringConn(conn, retv + 1, s, len, &error_code);
+    if (error_code > 0)
+    {
+        len_esc = 0;
+    }
+    retv[len_esc + 1] = '\'';
+    retv[len_esc + 2] = '\0';
+
+    return retv;
+}
+
+// helper function to collect schemas from search_path
+std::vector<std::string> get_schema_names(PGconn * conn)
+{
+    std::vector<std::string> schema_names;
+    PGresult* search_path = PQexec(conn, "SHOW search_path");
```

Comment:
Please use `details::postgresql_result` to avoid having explicit `PQclear()` calls in the code.